overlord/hookstate/context.go

```diff
@@ -127,3 +128,50 @@ func (c *Context) Get(key string, value interface{}) error {
 
 	return nil
 }
+
+// State returns the state contained within the context
+func (c *Context) State() *state.State {
+	return c.task.State()
+}
+
+// Cached returns the cached value associated with the provided key.
+// It returns nil if there is no entry for key.
+func (c *Context) Cached(key interface{}) interface{} {
+	contextCache := c.task.State().Cached("hook-context")
```

Comment:
This is actually bogus, as it means every single hook will end up with the same cached state, and thus the same config transaction, etc. We want these independent, and we don't need to use State() here, as there's no benefit in making these values global. The suggestion was actually to copy the logic out of state.go, reproducing the same logic here. Sorry for not being clear.

Who would remove these values otherwise?